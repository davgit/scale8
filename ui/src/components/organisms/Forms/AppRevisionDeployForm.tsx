import { FC } from 'react';
import {
    AppRevisionDeployFormProps,
    AppRevisionDeployValues,
} from '../../../utils/forms/AppRevisionDeployDialogFormUtils';
import { Box, DialogContent } from '@mui/material';
import DialogActionsWithCancel from '../../molecules/DialogActionsWithCancel';
import FormFlex from '../../atoms/FormFlex';
import { DialogFormSelect } from '../../atoms/DialogFormInputs/DialogFormSelect';
import { DialogFormContextProvider } from '../../../context/DialogFormContext';

const EnvironmentSelect: FC<AppRevisionDeployFormProps> = (props: AppRevisionDeployFormProps) => {
    const notAvailable = props.availableEnvironments.length < 1;

    if (notAvailable) {
        return <small>Already deployed to all available environments.</small>;
    }

    return (
        <DialogFormSelect
            label="Environment"
            name="environmentId"
            values={props.availableEnvironments}
        />
    );
};

const AppRevisionDeployForm: FC<AppRevisionDeployFormProps> = (
    props: AppRevisionDeployFormProps,
) => {
    const notAvailable = props.availableEnvironments.length < 1;

    return (
        <DialogFormContextProvider<AppRevisionDeployValues> formProps={props}>
            <Box
                sx={{
                    minWidth: '400px',
                    padding: (theme) => theme.spacing(0, 2, 1, 2),
                }}
            >
                <FormFlex handleSubmit={props.handleSubmit}>
                    <DialogContent>
                        <EnvironmentSelect {...props} />
                    </DialogContent>
                    <DialogActionsWithCancel
                        disableSubmit={props.isSubmitting || notAvailable}
                        handleDialogClose={props.handleDialogClose}
                        confirmText={props.submitText}
                        ignoreChanges
                    />
                </FormFlex>
            </Box>
        </DialogFormContextProvider>
    );
};

export default AppRevisionDeployForm;
