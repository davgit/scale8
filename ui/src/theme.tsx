import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
