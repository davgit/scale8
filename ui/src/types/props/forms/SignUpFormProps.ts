import { FormProps } from '../../../hooks/form/useFormValidation';
import { ApolloError } from '@apollo/client';
import { RefObject } from 'react';
import HCaptcha from '@hcaptcha/react-hcaptcha';

export type SignUpValues = {
    newPassword: string;
    newPasswordConfirm: string;
    domain: string;
    orgName: string;
    fullName: string;
    email: string;
    agree: boolean;
    CAPTCHAToken: string;
    tempAccessCode: string;
};

export type SignUpFormProps = FormProps<SignUpValues> & {
    gqlError?: ApolloError;
    success: boolean;
    email?: string;
    type: string;
    target: string | undefined;
    loading: boolean;
    qsEmail: string | undefined;
    captcha: RefObject<HCaptcha>;
};
