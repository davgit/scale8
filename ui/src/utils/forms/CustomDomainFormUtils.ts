import { ValidateConfiguration } from '../validators/validateFormValues';
import domainValidator from '../validators/domainValidator';
import { FormProps } from '../../hooks/form/useFormValidation';

export type CustomDomainValues = {
    domain: string;
    certificate: string;
    key: string;
};

export const CustomDomainValidators: ValidateConfiguration<CustomDomainValues>[] = [
    {
        field: 'domain',
        validator: domainValidator,
        error: () => 'Invalid domain',
        ignoreEmpty: true,
    },
];

export type CustomDomainFormProps = FormProps<CustomDomainValues> & {
    cname: string;
};
