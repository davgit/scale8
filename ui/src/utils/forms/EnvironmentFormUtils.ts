import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import urlValidator from '../validators/urlValidator';

export type EnvVariable = { key: string; value: string };

export type EnvironmentValues = {
    name: string;
    url: string;
    revisionId: string;
    variables?: EnvVariable[];
    comments: string;
};

const EnvironmentCreateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

const EnvironmentUpdateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

export { EnvironmentCreateValidators, EnvironmentUpdateValidators };
