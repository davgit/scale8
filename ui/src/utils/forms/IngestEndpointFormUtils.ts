import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import { StorageProviderFields, storageProviderValidators } from '../StorageProviderUtils';
import { IngestSchemaWizard } from '../../gql/generated/globalTypes';

export type IngestEndpointValues = StorageProviderFields & {
    name: string;
    wizard: IngestSchemaWizard | '';
    analyticsEnabled: boolean;
};

const IngestEndpointValidators: ValidateConfiguration<IngestEndpointValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Ingest Endpoint name too short',
    },
    ...storageProviderValidators,
];

export { IngestEndpointValidators };
