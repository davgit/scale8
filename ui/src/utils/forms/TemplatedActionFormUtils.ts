import { PlatformDataMapInput } from '../../types/DataMapsTypes';
import { PlatformActionPermissionInput } from '../../types/ActionPermissionsTypes';
import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import descriptionValidator from '../validators/descriptionValidator';
import requiredStringValidator from '../validators/requiredStringValidator';
import parsedCodeValidator from '../validators/parsedCodeValidator';
import { PlatformActionPermissionRequest } from '../../gql/generated/globalTypes';

export type TemplatedActionsValues = {
    name: string;
    description: string;
    icon: string;
    code: string;
    execRaw: boolean;
    platformDataMaps: PlatformDataMapInput[];
    permissionRequests: PlatformActionPermissionInput[];
};

const TemplatedActionValidators: ValidateConfiguration<TemplatedActionsValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Action name too short',
    },
    {
        field: 'description',
        validator: descriptionValidator,
        error: () => 'Description too short',
    },
    {
        field: 'name',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: parsedCodeValidator,
        error: (result) => `Invalid Code${result === 0 || `, Error: ${result}`}`,
    },
    {
        field: 'code',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as string).includes('success()') ? -1 : 0;
        },
        error: () => 'Action code must invoke "success()"',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'description',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'permissionRequests',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as PlatformActionPermissionInput[]).find((_) => {
                if (
                    _.active &&
                    (_.permission === PlatformActionPermissionRequest.READ_PAGE_URL ||
                        _.permission === PlatformActionPermissionRequest.READ_REFERRER_URL)
                ) {
                    return _.urlParts === undefined || _.urlParts.length === 0;
                }
                return false;
            }) === undefined
                ? -1
                : 0;
        },
        error: () => 'Select at least one.',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
];

export { TemplatedActionValidators };
