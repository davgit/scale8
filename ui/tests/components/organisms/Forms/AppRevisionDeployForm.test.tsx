import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import AppRevisionDeployForm from '../../../../src/components/organisms/Forms/AppRevisionDeployForm';
import { AppRevisionDeployValues } from '../../../../src/utils/forms/AppRevisionDeployDialogFormUtils';

const MockForm: FC = () => {
    const formProps = useMockFormProps<AppRevisionDeployValues>({
        environmentId: '',
    });

    return <AppRevisionDeployForm {...formProps} availableEnvironments={[]} />;
};

describe('AppRevisionDeployForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
