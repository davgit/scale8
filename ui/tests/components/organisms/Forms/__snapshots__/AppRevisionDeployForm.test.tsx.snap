// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRevisionDeployForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1r24ujm"
  >
    <form
      class="MuiBox-root css-mjj15b"
    >
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <small>
          Already deployed to all available environments.
        </small>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-snbeu5-MuiDialogActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-7kx4b6-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
